### ISSUE TITLE
[Feature request] (Android/iOS) Allow tapping screen to turn page in e-reader

### ISSUE THREAD

**Issue by nerdpersonthing** [advplyr/audiobookshelf-app#800]
Currently, when using the e-reader mode to read (non-audio) e-books in the Android and iOS apps, the only way to turn the page forward or back is using a swipe gesture.

In most e-reader apps, as well as most e-reader devices (Kindle, Nook, etc), simply tapping different regions of the screen also works to turn the page in the same way. As an example, see the attached screenshot from Calibre's webserver.

![Screenshot_20230721-172538](https://github.com/advplyr/audiobookshelf-app/assets/24740711/4efdf5fe-27a8-46ac-b1b3-19b1f91b8c3e)

It would be great if this could be implemented in Audiobookshelf for accessibility and ease of use. 

One potential implementation that would make sense to me would be adding an option in the "Ereader Settings" menu - a selector labeled "Turn pages with:", with options "Tap", "Swipe", or "Both". In my opinion, "Both" should be made the default, as this is the behavior in the vast majority of other e-readers I have used.

**Comment by advplyr** [comment]
Have you seen settings in other ereaders that allow you to change how the page navigation works?

**Comment by nerdpersonthing** [comment]
Sure, here are a few examples.

Calibre doesn't actually have any options to configure its page navigation, to my knowledge - it just functions with both tap and swipe, and doesn't seem to have any way to customize that behavior.

Meanwhile, ReadEra (my preferred document reader on Android) does have some pretty extensive options to customize how its page navigation works - see the screenshots below.

![Screenshot_20230721-175046](https://github.com/advplyr/audiobookshelf-app/assets/24740711/dfc9e302-806c-4f38-a0e1-34e188e3885c)

(It refers to navigation by tapping as "Flipping by touch.")
![Screenshot_20230721-175004](https://github.com/advplyr/audiobookshelf-app/assets/24740711/e4ae7498-6dce-47d2-866e-4eba861eeb44)

Scheme 1:
![Screenshot_20230721-175637](https://github.com/advplyr/audiobookshelf-app/assets/24740711/c2ba7290-a6d0-4616-82f9-8cc41bc3b168)

Scheme 2:
![Screenshot_20230721-175029](https://github.com/advplyr/audiobookshelf-app/assets/24740711/99925129-2d8a-4882-8c43-5acc17b753ac)

Scheme 3:
![Screenshot_20230721-175035](https://github.com/advplyr/audiobookshelf-app/assets/24740711/04db0d3d-8bbf-4e08-a26f-1331f572965d)

I don't think customization this extensive needs to be in audiobookshelf; merely being able to enable/disable navigation that functions like Scheme 1, or Calibre's layout from the original post, would be sufficient IMO. (Though having the option to navigate via volume buttons as well would be nice.)

**Comment by KaiStarkk** [comment]
Agreed that detailed customization isn't needed, but a tap option would be very helpful as it prevents ghosting and works better on devices with lesser CPU / refresh rate.

**Comment by dathbe** [comment]
I support this. I would say that I think the later mock-ups in this thread are better than the on in the original post because most ereaders I’m familiar with will bring up the settings/options when you tap in the middle, but will turn the page if you have the edge of the screen.

**Comment by KaiStarkk** [comment]
Appears to be possible in upstream, although their docs are down. Helpful references:
https://github.com/futurepress/epub.js/issues/46#issuecomment-530419388
https://github.com/futurepress/epub.js/issues/1400

**Comment by KaiStarkk** [comment]
Just noticed, the default behaviour in-app is now to turn the page instantly. Not sure if anyone feels this issue is still necessary? This has addressed my major concern of power draw and ghosting.

**Comment by nerdpersonthing** [comment]
Just tested it again, that doesn't appear to be the case for me on Android 0.9.76. Tapping the page still brings up the menu, with seemingly no option to instantly turn the page.

**Comment by KaiStarkk** [comment]
To clarify it used to be that there was a page turn animation which was the
crux of the issue for me — you’re right that it doesn’t page on a tap, but
it does now instantly page instead of animating which is an improvement
from the power and e-ink standpoint

**Comment by dathbe** [comment]
Animation was not my issue. By far the most common interaction I have while reading (I suspect others too) is turning a page. That should be as simple as possible a gesture. Nothing is simpler than a single tap with a single finger. Instead, that brings up a menu I rarely need.  And to turn the page I have to put the finger down, move it, then pick it up.  So the request is regions on the screen so that menu only comes up if, say, you tap the top, bottom, or sent of screen. Tapping left side will turn page back. Tapping right will turn the page forward.  This is relatively standard reader behavior in my experience.  But, to reiterate, it’s not secondary effects like battery life. 

I’ve found the abs is not consistent about saving my reading position with books, so I’ve switched back to kavita for my e-reading (it does allow single tap page turning, too). I use abs exclusively for audiobooks now. But will give it another run when some of the kinks are worked out.

**Comment by nerdpersonthing** [comment]
> @dathbe: Animation was not my issue. By far the most common interaction I have while reading (I suspect others too) is turning a page. That should be as simple as possible a gesture. Nothing is simpler than a single tap with a single finger. Instead, that brings up a menu I rarely need. And to turn the page I have to put the finger down, move it, then pick it up. So the request is regions on the screen so that menu only comes up if, say, you tap the top, bottom, or center of screen. Tapping left side will turn page back. Tapping right will turn the page forward. This is relatively standard reader behavior in my experience. But, to reiterate, it’s not secondary effects like battery life.

Agreed - my issue here was much more with the touch gestures than the animations. It's cumbersome and awkward to navigate by swiping, and in my opinion, tap-to-turn-page should be enabled by default. At a bare minimum, it should be an option.

> @KaiStarkk: To clarify it used to be that there was a page turn animation which was the crux of the issue for me — you’re right that it doesn’t page on a tap, but it does now instantly page instead of animating which is an improvement from the power and e-ink standpoint

Curious - for me, the page turn animation does still seem to be present on v0.9.77-beta, on both of my Android devices (a Moto 5G Ace and a Boox Page).

Regardless, I think having an option to disable animations entirely would be great (for e-ink devices and others), and it could also apply across the app, not just in the ebook interface. This is outside the scope of this issue, though, so I've created a new issue for it (https://github.com/advplyr/audiobookshelf-app/issues/1398).

**Comment by nerdpersonthing** [comment]
Also, it has been over a year since I opened this request - @advplyr, may I ask what the status of this feature is, and if there are currently any plans to implement it? (Either way, I do understand, and I appreciate all the time you already put into this awesome piece of software.)

If not, I'd be happy to take a crack at trying to add it myself, but it would be my first time contributing to a major GitHub project and I may have to learn the ropes a bit :)

**Comment by weilBaum9** [comment]
Are there any updates regarding this feature?
